editor/plugins/control_editor_plugin.h

```diff
@@ -127,123 +128,107 @@ class EditorInspectorPluginControl : public EditorInspectorPlugin {
+	GDCLASS(ControlAnchorPresetPicker, MarginContainer);
 
-	UndoRedo *undo_redo;
-	EditorSelection *editor_selection;
+	Map<LayoutPreset, Button *> preset_buttons;
 
-	enum MenuOption {
-		ANCHORS_AND_OFFSETS_PRESET_TOP_LEFT,
-		ANCHORS_AND_OFFSETS_PRESET_TOP_RIGHT,
-		ANCHORS_AND_OFFSETS_PRESET_BOTTOM_LEFT,
-		ANCHORS_AND_OFFSETS_PRESET_BOTTOM_RIGHT,
-		ANCHORS_AND_OFFSETS_PRESET_CENTER_LEFT,
-		ANCHORS_AND_OFFSETS_PRESET_CENTER_RIGHT,
-		ANCHORS_AND_OFFSETS_PRESET_CENTER_TOP,
-		ANCHORS_AND_OFFSETS_PRESET_CENTER_BOTTOM,
-		ANCHORS_AND_OFFSETS_PRESET_CENTER,
-		ANCHORS_AND_OFFSETS_PRESET_TOP_WIDE,
-		ANCHORS_AND_OFFSETS_PRESET_LEFT_WIDE,
-		ANCHORS_AND_OFFSETS_PRESET_RIGHT_WIDE,
-		ANCHORS_AND_OFFSETS_PRESET_BOTTOM_WIDE,
-		ANCHORS_AND_OFFSETS_PRESET_VCENTER_WIDE,
-		ANCHORS_AND_OFFSETS_PRESET_HCENTER_WIDE,
-		ANCHORS_AND_OFFSETS_PRESET_WIDE,
-
-		ANCHORS_AND_OFFSETS_PRESET_KEEP_RATIO,
-
-		ANCHORS_PRESET_TOP_LEFT,
-		ANCHORS_PRESET_TOP_RIGHT,
-		ANCHORS_PRESET_BOTTOM_LEFT,
-		ANCHORS_PRESET_BOTTOM_RIGHT,
-		ANCHORS_PRESET_CENTER_LEFT,
-		ANCHORS_PRESET_CENTER_RIGHT,
-		ANCHORS_PRESET_CENTER_TOP,
-		ANCHORS_PRESET_CENTER_BOTTOM,
-		ANCHORS_PRESET_CENTER,
-		ANCHORS_PRESET_TOP_WIDE,
-		ANCHORS_PRESET_LEFT_WIDE,
-		ANCHORS_PRESET_RIGHT_WIDE,
-		ANCHORS_PRESET_BOTTOM_WIDE,
-		ANCHORS_PRESET_VCENTER_WIDE,
-		ANCHORS_PRESET_HCENTER_WIDE,
-		ANCHORS_PRESET_WIDE,
-
-		// Offsets Presets are not currently in use.
-		OFFSETS_PRESET_TOP_LEFT,
-		OFFSETS_PRESET_TOP_RIGHT,
-		OFFSETS_PRESET_BOTTOM_LEFT,
-		OFFSETS_PRESET_BOTTOM_RIGHT,
-		OFFSETS_PRESET_CENTER_LEFT,
-		OFFSETS_PRESET_CENTER_RIGHT,
-		OFFSETS_PRESET_CENTER_TOP,
-		OFFSETS_PRESET_CENTER_BOTTOM,
-		OFFSETS_PRESET_CENTER,
-		OFFSETS_PRESET_TOP_WIDE,
-		OFFSETS_PRESET_LEFT_WIDE,
-		OFFSETS_PRESET_RIGHT_WIDE,
-		OFFSETS_PRESET_BOTTOM_WIDE,
-		OFFSETS_PRESET_VCENTER_WIDE,
-		OFFSETS_PRESET_HCENTER_WIDE,
-		OFFSETS_PRESET_WIDE,
-
-		CONTAINERS_H_PRESET_FILL,
-		CONTAINERS_H_PRESET_FILL_EXPAND,
-		CONTAINERS_H_PRESET_SHRINK_BEGIN,
-		CONTAINERS_H_PRESET_SHRINK_CENTER,
-		CONTAINERS_H_PRESET_SHRINK_END,
-		CONTAINERS_V_PRESET_FILL,
-		CONTAINERS_V_PRESET_FILL_EXPAND,
-		CONTAINERS_V_PRESET_SHRINK_BEGIN,
-		CONTAINERS_V_PRESET_SHRINK_CENTER,
-		CONTAINERS_V_PRESET_SHRINK_END,
-	};
+	static constexpr int grid_separation = 0;
+
+	void _add_row_button(HBoxContainer *p_row, const LayoutPreset p_preset, const String &p_name);
+	void _add_separator(BoxContainer *p_box, Separator *p_separator);
+	void _preset_button_pressed(const LayoutPreset p_preset);
+
+protected:
+	void _notification(int p_notification);
+	static void _bind_methods();
+
+public:
+	ControlAnchorPresetPicker();
+};
 
-	TextureRect *anchor_layouts_icon;
-	MenuButton *anchor_presets_menu;
-	PopupMenu *anchors_popup;
-	TextureRect *container_layouts_icon;
-	MenuButton *container_h_presets_menu;
-	MenuButton *container_v_presets_menu;
+class ControlAnchorsEditorToolbar : public VBoxContainer {
```

Comment:
I noticed it just now, but this class isn't a toolbar anymore, so the name could be changed.
Same with the other class.

Reply:
I don't have a better name in mind at the moment, and it can still be considered a (vertical) toolbar. Should be trivial to change later, as this is not exposed anywhere.